src/jsx.ts

```diff
@@ -101,7 +102,35 @@ export function jsx(
 // See https://www.typescriptlang.org/docs/handbook/jsx.html#type-checking
 export namespace jsx {
   export type Element = VNode;
-  export interface IntrinsicElements {
+
+  /*
+  IfEquals/WritableKeys: https://stackoverflow.com/questions/52443276/how-to-exclude-getter-only-properties-from-type-in-typescript/52473108#52473108
+  */
+  export type IfEquals<X, Y, A = X, B = never> = (<T>() => T extends X
```

Comment:
I would prefer this type without the defaults. The first default is never used and I think no defaults make things a bit clearer and more explicit.

Reply:
Good points. Because it is StackOverflow copypasta intended for more usages I left it as-is, but it might be good to simplify it.